src/features/call/hooks/useCallWithTarget.ts

```diff
@@ -0,0 +1,24 @@
+import useRemoteItem from 'core/hooks/useRemoteItem';
+import { ZetkinCall } from '../types';
+import { currentCallLoad, currentCallLoaded } from '../store';
+import { useApiClient, useAppSelector } from 'core/hooks';
+
+export default function useCallWithTarget(
+  orgId: number,
+  assignmentId: number
+): ZetkinCall {
+  const apiClient = useApiClient();
+  const callItem = useAppSelector((state) => state.call.currentCall);
+
+  const callWithTarget = useRemoteItem(callItem, {
+    actionOnLoad: () => currentCallLoad(),
+    actionOnSuccess: (data) => currentCallLoaded(data),
+    loader: () =>
+      apiClient.post<ZetkinCall>(
+        `/api/orgs/${orgId}/call_assignments/${assignmentId}/queue/head`,
+        {}
+      ),
+  });
```

Comment:
Is this what I overheard you talking about yesterday, that this hook should not do this `POST` but rather get the current call from somewhere else? (if not just ignore)

Reply:
We need this post call. The problem was to don't call the hook in a proper action, just call it when the component was rendered, so I will ignore the comment for now :)